ytsync/ytsync.go

```diff
@@ -343,7 +351,7 @@ func (s *Sync) stopAndUploadWallet(e *error) {
 			err := s.uploadWallet()
 			if err != nil {
 				if *e == nil {
-					e = &err
+					e = &err //not 100% sure
```

Comment:
why does this function return a pointer to an error?

Reply:
because the function is run as a deferred and needs to read/write to the error being returned by the original function. Is this a bad pattern?

Reply:
well actually to be correct it's not returning a pointer to an error, it's accepting a pointer to an error so that I can change it from within.